### ISSUE TITLE
Empty SVM line makes DD dump

### ISSUE THREAD

**Issue by YaYaB** [beniz/deepdetect#343]
I have an issue in prediction when I put an out of bounds svm index in the CURL request.
It makes DD dump.

#### Configuration
- Version of DeepDetect:
    - Locally compiled on Ubuntu 16.04 LTS
- Commit (shown by the server when starting):
    - Branch `Master`, 24f40b6993dedbedd223ced45f3b88dd0762da63
- GPUS:
    - 1 x Nvidia GTX Titan X (Maxwell)

#### Your question / the problem you're facing:
I have an issue in prediction when I put an out of bounds svm index in the CURL request. DD core dumps in that case.

First I launch DeepDetect
INPUT
```
INPUT
./dede --port 8080
```
OUTPUT
```
DeepDetect [ commit 95f235e4f9c9b827e434a65854b962e49b533f3f ]
INFO - 07:35:16 - Running DeepDetect HTTP server on localhost:8080
```

Second, I create the service that I name svm_test.
INPUT
``
curl -X PUT 'http://localhost:8080/services/svm_test' -d '{"mllib":"caffe","description":"svm_test classifier","type":"supervised","parameters":{"input":{"connector":"svm","characters":true},"mllib":{"nclasses":2}},"model":{"repository":"/home/svm_test/dd_models/model_svm"}}'
```

OUTPUT
```
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0824 07:38:43.746436 21717 svminputfileconn.cc:229] loaded SVM vocabulary of size=20147

INFO - 07:38:49 - Initializing net from parameters: 

INFO - 07:38:49 - Creating layer / name=inputl / type=MemorySparseData
INFO - 07:38:49 - Creating Layer inputl
INFO - 07:38:49 - inputl -> data
INFO - 07:38:49 - inputl -> label
INFO - 07:38:49 - Setting up inputl
INFO - 07:38:49 - Top shape: 1 20146 (20146)
INFO - 07:38:49 - Top shape: 1 (1)
INFO - 07:38:49 - Memory required for data: 80588
INFO - 07:38:49 - Creating layer / name=fc_data / type=SparseInnerProduct

...
```

Then I try to make a prediction with a valid index between 0 and 20146.
```
curl -X POST "http://localhost:8080/predict" -d "{\"service\":\"svm_test\",\"data\":[\"20146:0.18\"]}"
```

It works and I obtain the following:
```
{"status":{"code":200,"msg":"OK"},"head":{"method":"/predict","service":"svm_test","time":15869.0},"body":{"predictions":[{"uri":"0","classes":[{"prob":0.537948727607727,"last":true,"cat":"0"}]}]}}
```

However if I try and index out of the bounds (here 20147) I get an error:
```
curl -X POST "http://localhost:8080/predict" -d "{\"service\":\"svm_test\",\"data\":[\"20147:0.18\"]}" 
```

The reply is the following:
```
curl: (52) Empty reply from server
```

DD's logs print:
```
[1]    21704 floating point exception (core dumped)  ./dede --port 8080
```

If a try another index out of the bounds (here -1):
```
curl -X POST "http://localhost:8080/predict" -d "{\"service\":\"svm_test\",\"data\":[\"-1:0.18\"]}" 
```

The reply is the following:
```
curl: (52) Empty reply from server
```

DD's logs print:
```
[1]    21787 floating point exception (core dumped)  ./dede --port 8080
```

**Comment by beniz** [comment]
Thanks for the report! I believe the out of bound index is correctly handled by DD, but that the resulting empty line is not, as suggested by our attached PR #344.

Can you confirm that 

```
curl -X POST "http://localhost:8080/predict" -d "{"service":"svm_test","data":["20146:0.18 20147:0.18"]}"
```

works correctly even without your PR ?

**Comment by YaYaB** [comment]
Yes, I confirm that it works even without the PR!

**Comment by beniz** [comment]
OK, thanks.

**Comment by beniz** [comment]
Fixed by #344 .
